src/library/filescanner.c

```diff
@@ -1816,12 +1816,62 @@ playlist_add_path(FILE *fp, int pl_id, const char *path)
   return 0;
 }
 
+static int
+ensure_file_record(const char *path)
+{
+  struct query_params qp;
+  struct db_media_file_info dbmfi;
+  struct media_file_info mfi;
+  bool found;
+  int ret;
+
+  memset(&qp, 0, sizeof(struct query_params));
+  qp.type = Q_ITEMS;
+  qp.sort = S_ARTIST;
+  qp.idx_type = I_NONE;
+  qp.filter = db_mprintf("(f.path = %Q)", path);
+
+  ret = db_query_start(&qp);
+  if (ret < 0)
+    {
+      db_query_end(&qp);
+      free(qp.filter);
+      return -1;
+    }
+
+  found = ((ret = db_query_fetch_file(&qp, &dbmfi)) == 0) && (dbmfi.id);
```

Comment:
This check is not necessary, if we make sure to call this method only if the path is not already in the files table (see other comment for `queue_save`). If we want to play it safe, better to use `db_file_id_bypath`.